gammapy/background/energy_offset_array.py

```diff
@@ -95,12 +98,107 @@ def plot_image(self, ax=None, offset=None, energy=None, **kwargs):
         ax.legend()
-        image = ax.imshow(self.data, extent=extent, **kwargs)
+        image = ax.imshow(self.data.value, extent=extent, **kwargs)
         plt.colorbar(image)
         return ax
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyOffsetArray` from  fits file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdu_list = fits.open(filename)
+        hdu = hdu_list[1]
+        return cls.from_fits(hdu)
+
+    @classmethod
+    def from_fits(cls, hdu):
+        """Read `EnergyOffsetArray` from a fits binary table.
+
+        Parameters
+        ----------
+        hdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the EnergyOffsetArray.
+
+        Returns
+        -------
+        cube : `~gammapy.background.EnergyOffsetArray`
+            EnergyOffsetArray object.
+        """
+
+        header = hdu.header
+        data = hdu.data
+        # get offset and energy binning
+        offset_edges = _make_bin_edges_array(data['THETA_LO'].squeeze(), data['THETA_HI'].squeeze())
+        energy_edges = _make_bin_edges_array(data['ENERG_LO'].squeeze(), data['ENERG_HI'].squeeze())
+
+        # get data
+        energy_offset_array = data['EnergyOffsetArray'].squeeze()
+        return cls(energy_edges, offset_edges, energy_offset_array)
+
+    def to_table(self):
+        """Convert `EnergyOffsetArray` to astropy table format.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table containing the EnergyOffsetArray.
+        """
+        # table
+        table = Table()
+        table['THETA_LO'] = [self.offset[:-1]]
+        table['THETA_HI'] = [self.offset[1:]]
+        table['ENERG_LO'] = [self.energy[:-1]]
+        table['ENERG_HI'] = [self.energy[1:]]
+        table['EnergyOffsetArray'] = [Quantity(self.data, " u ")]
+        table.meta['name'] = 'TO DEFINE'
+
+        return table
+
+    def to_fits(self):
+        """Convert `EnergyOffsetArray` to binary table fits format.
+
+        Returns
+        -------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            Table containing the EnergyOffsetArray.
+        """
+        return table_to_fits_table(self.to_table())
+
+    def write(self, filename, **kwargs):
+        """ Write EnergyOffsetArray to fits file"""
+        self.to_fits().writeto(filename, **kwargs)
+
+    def evaluate(self, energy, offset, interpolate_params):
+        """
+        Interpolate the value of the `EnergyOffsetArray` at a given offset and Energy
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+         energy value
+        offset : `~astropy.coordinates.Angle`
+        offset value
+        interpolate_params: dict
+        give the options for the RegularGridInterpolator
+
+        Returns
+        -------
+        Interpolated value
+        """
+        from scipy import interpolate
```

Comment:
yes I push the latest version
Le 12 févr. 2016 à 15:05, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/background/energy_offset_array.py:
> 
> > +        Interpolate the value of the `EnergyOffsetArray` at a given offset and Energy
> > +
> > +        Parameters
> > +        ----------
> > +        energy : `~astropy.units.Quantity`
> > +         energy value
> > +        offset : `~astropy.coordinates.Angle`
> > +        offset value
> > +        interpolate_params: dict
> > +        give the options for the RegularGridInterpolator
> > +
> > +        Returns
> > +        -------
> > +        Interpolated value
> > +        """
> > +        from scipy import interpolate
> Did you push the latest version?
> Is this ready to be merged?
> 
> I'd like to merge this now, then rebase #449 and merge that one.
> This way you won't have to deal with merge conflicts...
> 
> —
> Reply to this email directly or view it on GitHub.
>